packages/flutter_tools/templates/create/ios.tmpl/Runner/AppDelegate.m

```diff
@@ -14,25 +14,34 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
+  // Sent when the application is about to move from active to inactive state. This can occur for certain types of temporary interruptions (such as an incoming phone call or SMS message) or when the user quits the application and it begins the transition to the background state.
+  // Use this method to pause ongoing tasks, disable timers, and throttle down OpenGL ES frame rates. Games should use this method to pause the game.
 }
 
 - (void)applicationDidEnterBackground:(UIApplication *)application {
-    // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
-    // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+  // Use this method to release shared resources, save user data, invalidate timers, and store enough application state information to restore your application to its current state in case it is terminated later.
+  // If your application supports background execution, this method is called instead of applicationWillTerminate: when the user quits.
+
+  // The following keeps the Flutter session alive when the device screen locks
+  // in debug mode. It allows continued use of features like hot reload and 
+  // taking screenshots once the device unlocks again.
+  if ([@"debug" isEqualToString:
+      [[NSBundle mainBundle] objectForInfoDictionaryKey:@"FLTBuildMode"]]) {
+    [application beginBackgroundTaskWithName:@"Flutter debug task"
+                           expirationHandler:^{NSLog(@"Flutter debug task expired");}];
+  }
 }
 
 - (void)applicationWillEnterForeground:(UIApplication *)application {
-    // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+  // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
 }
 
 - (void)applicationDidBecomeActive:(UIApplication *)application {
-    // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+  // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
 }
 
 - (void)applicationWillTerminate:(UIApplication *)application {
-    // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
+  // Called when the application is about to terminate. Save data if appropriate. See also applicationDidEnterBackground:.
 }
```

Comment:
Since this code will be part of the developer's codebase, not Flutters, I'd stick with the defaults that Xcode provides (4 space indent) as opposed to Flutter style.